_2020/shell-tools.md

```diff
@@ -102,7 +102,7 @@ echo "Starting program at $(date)" # Date will be substituted
 echo "Running program $0 with $# arguments with pid $$"
 
 for file in "$@"; do
-    grep foobar "$file" > /dev/null 2> /dev/null
+    grep foobar "$file" &> /dev/null
```

Comment:
While this works and is indeed more concise than redirecting separately, I personally would rewrite this in two ways:

- Use `grep -q` so there is no output to begin with
- Check the exit status directly instead of looking at `$?`

```sh
if ! grep -q foobar "$file"; then
    echo "File $file does not have any foobar, adding one"
    echo "# foobar" >> "$file"
fi
```

This would also require to update the next paragraph; reference: [Bash Pitfall 44](https://mywiki.wooledge.org/BashPitfalls#cmd.3B_.28.28_.21_.24.3F_.29.29_.7C.7C_die).